src/com/google/enterprise/adaptor/prebuilt/DateFilter.java

```diff
@@ -0,0 +1,384 @@
+ * <p>
+ * {@code days} The cut-off for the date value. Document's whose date
+ * value is more than {@code days} before present will not be indexed.
+ * This can be used to have a rolling window of indexed documents. Those
+ * whose date have expired will be removed from the index.  Only one of
+ * {@code date} or {@code days} configuration may be specified.
+ * <p>
+ * {@code keyset} The location of the date value to consider. The
+ * {@code keyset} may be set to {@code metadata} or to {@code params}
+ * to restrict the search to metadata or params, respectively.
+ *  If {@code keyset} is not specified, it defaults to {@code metadata}.
+ * <p>
+ * Example 1: skip documents that have not been accessed for more than 3 years:
+ * <pre><code>
+   metadata.transform.pipeline=dateFilter
+   metadata.transform.pipeline.dateFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.DateFilter.create
+   metadata.transform.pipeline.dateFilter.key=Last_Access_Date
+   metadata.transform.pipeline.dateFilter.days=1095
+   </code></pre>
+ * <p>
+ * Example 2: skip pre-Y2K client records that used old-style US date formats:
+ * <pre><code>
+   metadata.transform.pipeline=dateFilter
+   metadata.transform.pipeline.dateFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.DateFilter.create
+   metadata.transform.pipeline.dateFilter.key=Last_Visit_Date
+   metadata.transform.pipeline.dateFilter.format=MM/dd/YY
+   metadata.transform.pipeline.dateFilter.date=01/01/00
+   </code></pre>
+ * <p>
+ * Example 3: skip documents that have not been modified since 2010:
+ * <pre><code>
+   metadata.transform.pipeline=dateFilter
+   metadata.transform.pipeline.dateFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.DateFilter.create
+   metadata.transform.pipeline.dateFilter.keyset=params
+   metadata.transform.pipeline.dateFilter.key=Last-Modified-Millis-UTC
+   metadata.transform.pipeline.dateFilter.format=millis
+   metadata.transform.pipeline.dateFilter.date=2010-01-01
+   </code></pre>
+ */
+public class DateFilter implements MetadataTransform {
+  private static final Logger log
+      = Logger.getLogger(DateFilter.class.getName());
+
+  private static final String ISO_8601_FORMAT = "yyyy-MM-dd";
+
+  // TODO (bmj): remove this and use the one in MetadataTransforms.
+  public enum Keyset {
+    METADATA("metadata"),
+    PARAMS("params");
+
+    private final String name;
+
+    private Keyset(String name) {
+      this.name = name;
+    }
+
+    public static Keyset from(String val) {
+      return (val == null) ? METADATA : Keyset.valueOf(val.toUpperCase());
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+  };
+
+  /** The name of the key (either Metadata key or params key) to match. */
+  private final String key;
+
+  /**
+   * If {@code METADATA}, search the metadata for the specified key;
+   * if {@code PARAMS},  search the params for the specified key;
+   */
+  private Keyset keyset = Keyset.METADATA;
```

Comment:
As in RegexFilter, the create method will ensure that this is set to METADATA by default, so we can elide the value and make this field final.